Content.Shared/Psionics/PsionicComponent.cs

```diff
@@ -1,20 +1,137 @@
-using Content.Shared.Actions;
+using Content.Shared.Psionics;
 using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
 
 namespace Content.Shared.Abilities.Psionics
 {
     [RegisterComponent, NetworkedComponent]
     public sealed partial class PsionicComponent : Component
     {
-        public EntityUid? PsionicAbility = null;
-
         /// <summary>
         ///     Ifrits, revenants, etc are explicitly magical beings that shouldn't get mindbreakered.
         /// </summary>
-        [DataField("removable")]
+        [DataField]
         public bool Removable = true;
 
-        [DataField("activePowers")]
-        public HashSet<Component> ActivePowers = new();
+        /// <summary>
+        ///     The list of all powers currently active on a Psionic, by power Prototype.
+        ///     TODO: Not in this PR due to scope, but this needs to go to Server and not Shared.
+        /// </summary>
+        [DataField]
+        public HashSet<PsionicPowerPrototype> ActivePowers = new();
+
+        /// <summary>
+        ///     The list of each Psionic Power by action with entityUid.
+        /// </summary>
+        [DataField]
+        public List<(EntProtoId Id, EntityUid? Entity)> Actions = new();
```

Comment:
Why is this a list of tuples rather than a Dictionary?

Reply:
The actual reason for why, is that Actions are both an Id and an Entity, and not having both in the same place means that the Entity CANNOT in any way be tied to a Uid.